Source/Core/Core/HW/EXI_DeviceSlippi.cpp

```diff
@@ -21,8 +25,46 @@
 #include "Common/StringUtil.h"
 #include "Common/Thread.h"
 #include "Core/HW/Memmap.h"
+#include "Core/HW/SlippiPlayback.h"
 #include "Core/NetPlayClient.h"
 
+#include "Core/Core.h"
+#include "Core/CoreTiming.h"
+#include "Core/HW/SystemTimers.h"
+#include "Core/State.h"
+
+// state diffs keyed by frameIndex;
+static std::map<int32_t, std::string> diffsByFrame;
+
+// diffs are processed async
+std::vector<std::future<std::pair<int32_t, std::string>>> futureDiffs;
+bool g_shouldJumpBack = false;
```

Comment:
doing this with a bunch of global/externs rn for testing, happy to use another pattern